source/frontend/src/features/mapSideBar/property/tabs/propertyResearch/update/models.ts

```diff
@@ -41,65 +44,64 @@ export class UpdatePropertyFormModel {
-        ? 'yes'
-        : 'no';
-    model.isLegalOpinionObtained =
-      base.isLegalOpinionObtained === undefined
-        ? 'unknown'
-        : base.isLegalOpinionObtained
-        ? 'yes'
-        : 'no';
-    model.documentReference = base.documentReference;
-    model.researchSummary = base.researchSummary;
+    model.propertyName = base.propertyName ?? undefined;
+    model.displayOrder = base.displayOrder ?? undefined;
+    model.isLegalOpinionRequired = !exists(base.isLegalOpinionRequired)
+      ? 'unknown'
+      : base.isLegalOpinionRequired
+      ? 'yes'
+      : 'no';
+    model.isLegalOpinionObtained = !exists(base.isLegalOpinionObtained)
+      ? 'unknown'
+      : base.isLegalOpinionObtained
+      ? 'yes'
+      : 'no';
+    model.documentReference = base.documentReference ?? undefined;
+    model.researchSummary = base.researchSummary ?? undefined;
     model.propertyId = base.property?.id;
-    model.researchFileId = base.file?.id;
-    model.researchFileVersion = base.file?.rowVersion;
+    model.researchFile = base.file;
 
-    model.purposeTypes = base.purposeTypes?.map((x: Api_ResearchFilePropertyPurposeType) =>
+    model.purposeTypes = base.purposeTypes?.map((x: ApiGen_Concepts_PropertyPurpose) =>
       PropertyResearchFilePurposeFormModel.fromApi(x),
     );
-    model.rowVersion = base.rowVersion;
+    model.rowVersion = base.rowVersion ?? undefined;
     return model;
   }
 
-  public toApi(): Api_ResearchFileProperty {
+  public toApi(): ApiGen_Concepts_ResearchFileProperty {
     return {
-      id: this.id,
-      propertyId: this.propertyId,
-      propertyName: this.propertyName,
-      displayOrder: this.displayOrder,
+      id: this.id ?? 0,
+      propertyId: this.propertyId ?? 0,
+      propertyName: this.propertyName ?? null,
+      displayOrder: this.displayOrder ?? null,
       isLegalOpinionRequired:
-        this.isLegalOpinionRequired === undefined || this.isLegalOpinionRequired === 'unknown'
-          ? undefined
+        !exists(this.isLegalOpinionRequired) || this.isLegalOpinionRequired === 'unknown'
+          ? null
           : this.isLegalOpinionRequired === 'yes'
           ? true
           : false,
       isLegalOpinionObtained:
-        this.isLegalOpinionObtained === undefined || this.isLegalOpinionObtained === 'unknown'
-          ? undefined
+        !exists(this.isLegalOpinionObtained) || this.isLegalOpinionObtained === 'unknown'
+          ? null
           : this.isLegalOpinionObtained === 'yes'
           ? true
           : false,
-      documentReference: this.documentReference,
-      researchSummary: this.researchSummary,
-      property: { id: this.propertyId },
-      file: { id: this.researchFileId, rowVersion: this.researchFileVersion },
-      purposeTypes: this.purposeTypes?.map(x => x.toApi()),
-      rowVersion: this.rowVersion,
+      documentReference: this.documentReference ?? null,
+      researchSummary: this.researchSummary ?? null,
+      property: null,
```

Comment:
this doesn't seem passive.

Reply:
the backend only cares about the property id, In that light setting only that member makes more sense than creating a property that only contains that information. The mapper is aligned with that.